dev-docs/design/how-lit-html-works.md

```diff
@@ -0,0 +1,429 @@
+
+TTLs provide an explicit syntax for separating static strings of HTML content and dynamic JavaScript values. The static strings are immutable and preserve identity across multiple evaluations of the tagged literal making them ideal for use as a cache key.
+
+The `<template>` element provides a container for HTML content that can later be cloned into the document. In a template element: scripts don't run, styles don't apply, custom elements don't upgrade, etc. During rendering, a template's content can be cloned or imported into the Document which makes the cloned nodes active.
+
+# Rendering
+
+## Summary of lit-html rendering phases
+
+lit-html templates are only a description of the UI. They must be rendered with `render()` to affect the DOM. The **Define** and **Render** phases are controlled by the user of Lit, while **Prepare**, **Create**, and **Update** are internal phases.
+
+<ol>
+<li>
+
+**Define**: `` const ui = (title) => html`<h1>${title}</h1>` ``
+
+</li>
+<li>
+
+**Render**: `render(ui("Example Title"), document.body)`
+
+</li>
+<ol type="a">
+<li>
+
+**Prepare**: The first time render is called for each unique TTL, create a `<template>` element containing the static HTML and Lit `TemplatePart`s encoding a mapping for each dynamic JavaScript value to its location in the static HTML.
+
+</li>
+<li>
+
+**Create**: The first time render is called for a given node and template, clone the `<template>` contents into the DOM and instantiate the Lit `Part`s which manage the dynamic content.
+
+</li>
+
+<li>
+
+**Update**: Iterate over the dynamic JS values and associated Lit `Part` only committing the values that have changed to the DOM.
+
+</li>
+</ol>
+</ol>
+
+The end result is `<h1>Example Title</h1>` rendered in the `<body>` element. If the user now executes `render(ui("Updated"), document.body)`, only the **Update** phase runs skipping the **Prepare** and **Create** phases.
+
+## Motivating example
+
+This example will be referred to throughout the rest of the document to concretely show what lit-html is doing during each internal rendering phase.
+
+```html
+<!-- File: index.html -->
+<script type="module" src="./example.js"></script>
+<style>
+  .odd {
+    color: green;
+    text-decoration: underline;
+  }
+</style>
+
+<div id="container"></div>
+```
+
+```js
+// File: example.js
+import {html, render} from 'lit';
+
+const container = document.querySelector('#container');
+
+const counterUi = (count) => html` <span class="${count % 2 == 1 ? 'odd' : ''}">
+    ${count}
+  </span>
+  <button @click=${() => render(counterUi(count + 1), container)}>
+    Increment
+  </button>`;
+
+render(counterUi(0), container);
+```
+
+[Live example hosted on lit.dev.](https://lit.dev/playground/#gist=a1058bae08d79928fc2fafaf02b04581)
```

Comment:
Prefer "Live example in the playground"

"Playground" better communicates that they can mess with it, IMO

Reply:
Love it! Done!